nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/transforms/custom/Dilation2D.java

```diff
@@ -79,8 +82,34 @@ public Dilation2D(SameDiff sameDiff, SDVariable[] inputAndWeights, int[] strides
     public Dilation2D(INDArray[] inputArrays, INDArray[] outputs) {
         super(null, inputArrays, outputs);
 
+
+    }
+
+    public Dilation2D(@NonNull INDArray df, @NonNull INDArray weights, @NonNull int[] strides, @NonNull int[] rates, boolean isSameMode) {
+        super(null, new INDArray[]{df, weights},null);
+        Preconditions.checkArgument(rates.length == 4, "Dilation rate length must be 4, got an array with length %s with values %s", rates.length, rates)
+
+        this.isSameMode = isSameMode;
+
+        if (rates.length < 4)
+            throw new IllegalArgumentException("Dilation rate length must be 4.");
+        if (strides.length < 4)
+            throw new IllegalArgumentException("Strides length must be 4.");
```

Comment:
Now that we've added rates check (via Preconditions) we don't need another check via if statement.
We should change the strides check to the same sort of Preconditions check above and delete the if statement